src/java/org/apache/cassandra/index/sai/view/IndexViewManager.java

```diff
@@ -0,0 +1,163 @@
+     */
+    public Set<SSTableContext> update(Collection<SSTableReader> oldSSTables, Collection<SSTableContext> newSSTableContexts, boolean validate)
+    {
+        // Valid indexes on the left and invalid SSTable contexts on the right...
+        Pair<Set<SSTableIndex>, Set<SSTableContext>> indexes = context.getBuiltIndexes(newSSTableContexts, validate);
+
+        View currentView, newView;
+        Collection<SSTableIndex> newViewIndexes = new HashSet<>();
+        Collection<SSTableIndex> releasableIndexes = new ArrayList<>();
+        Collection<SSTableReader> toRemove = new HashSet<>(oldSSTables);
+        
+        do
+        {
+            currentView = view.get();
+            newViewIndexes.clear();
+            releasableIndexes.clear();
+
+            for (SSTableIndex sstableIndex : currentView)
+            {
+                // When aborting early open transaction, toRemove may have the same sstable files as newSSTableContexts,
+                // but different SSTableReader java objects with different start positions. So we need to release them
+                // from existing view.  see DSP-19677
+                SSTableReader sstable = sstableIndex.getSSTable();
+                if (toRemove.contains(sstable) || newViewIndexes.contains(sstableIndex))
+                    releasableIndexes.add(sstableIndex);
+                else
+                    newViewIndexes.add(sstableIndex);
+            }
+
+            for (SSTableIndex sstableIndex : indexes.left)
+            {
+                if (newViewIndexes.contains(sstableIndex))
+                    releasableIndexes.add(sstableIndex);
+                else
+                    newViewIndexes.add(sstableIndex);
+            }
+
+            newView = new View(context, newViewIndexes);
+        }
+        while (!view.compareAndSet(currentView, newView));
+
+        releasableIndexes.forEach(SSTableIndex::release);
+
+        if (logger.isTraceEnabled())
+            logger.trace(context.logMessage("There are now {} active SSTable indexes."), view.get().getIndexes().size());
+
+        return indexes.right;
+    }
+
+    public void drop(Collection<SSTableReader> sstablesToRebuild)
+    {
+        View currentView = view.get();
+
+        Set<SSTableReader> toRemove = new HashSet<>(sstablesToRebuild);
+        for (SSTableIndex index : currentView)
+        {
+            SSTableReader sstable = index.getSSTable();
+            if (!toRemove.contains(sstable))
+                continue;
+
+            index.markObsolete();
+        }
+
+        update(toRemove, Collections.emptyList(), false);
+    }
+
+    /**
+     * Called when index is dropped. Mark all {@link SSTableIndex} as released and per-column index files
+     * will be removed when in-flight queries completed and {@code obsolete} is true.
+     *
+     * @param obsolete true if index files should be deleted after invalidate; false otherwise.
+     */
+    public void invalidate(boolean obsolete)
```

Comment:
```suggestion
    public void invalidate(boolean delete)
```
nit: maybe in this class, just naming the argument around intentions makes sense

Reply:
This flag has been removed because we, currently, have no situation where the value was not `true`.